dissect/target/plugins/general/config.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from functools import lru_cache
+from typing import Optional, Union
+
+from dissect.target import Target
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.filesystems.config import (
+    ConfigurationEntry,
+    ConfigurationFilesystem,
+)
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.plugin import Plugin, internal
+
+
+class ConfigurationTreePlugin(Plugin):
+    __namespace__ = "config_tree"
+
+    def __init__(self, target: Target, dir_path: str = "/"):
+        super().__init__(target)
+        self.dir_path = dir_path
+        self.config_fs = None
+
+        target_dir_path = self.target.fs.path(dir_path)
+        if target_dir_path.is_dir():
+            self.config_fs = ConfigurationFilesystem(target, dir_path)
+
+    def check_compatible(self) -> None:
+        # This should be able to be retrieved, regardless of OS
+        if self.config_fs is None:
+            raise UnsupportedPluginError(f"{self.dir_path!r} could not be found.")
+        return None
+
+    @lru_cache(128)
```

Comment:
It's better to put something like this in the `__init__`:
```python
self.__call__ = lru_cache(128)(self.__call__)
```
I could look up the blog but you probably know the one I'm talking about anyway.